module/workflow/src/Domain/Entity/Workflow.php

```diff
@@ -77,79 +86,103 @@ public function getCode(): string
     }
 
     /**
-     * @param string $code
+     * @param StatusId $id
      *
      * @return bool
      */
-    public function hasStatus(string $code): bool
+    public function hasStatus(StatusId $id): bool
     {
-        return isset($this->statuses[$code]);
+        return isset($this->statuses[$id->getValue()]);
     }
 
     /**
-     * @param string $code
+     * @param StatusId $source
+     * @param StatusId $destination
      *
-     * @return Status
+     * @return bool
      */
-    public function getStatus(string $code): Status
+    public function hasTransition(StatusId $source, StatusId $destination): bool
     {
-        if (!$this->hasStatus($code)) {
-            throw  new \RuntimeException(sprintf('Status %s not exists', $code));
-        }
+        $key = $this->getTransitionKey($source, $destination);
 
-        return $this->statuses[$code];
+        return isset($this->transitions[$key]);
```

Comment:
Should be array_key_exists because isset check it with value and when value is equal NULL, false or 0 it will return false instead true

Reply:
There always be object